common/src/main/java/io/netty/util/concurrent/ScheduledFutureTask.java

```diff
@@ -25,14 +25,9 @@
 @SuppressWarnings("ComparableImplementedButEqualsNotOverridden")
 final class ScheduledFutureTask<V> extends PromiseTask<V> implements ScheduledFuture<V> {
     private static final AtomicLong nextTaskId = new AtomicLong();
-    private static final long START_TIME = System.nanoTime();
-
-    static long nanoTime() {
-        return System.nanoTime() - START_TIME;
-    }
 
     static long deadlineNanos(long delay) {
-        return nanoTime() + delay;
+        return System.nanoTime() + delay;
```

Comment:
i think we still have an issue. Adding the "expire" time to a "nanoTime" and checking it later is subject to overflow. I guess what we want is to always use `System.nanoTime()` as is and then compare the delta between the two snapshots to some expire interval.

```java
private static boolean isExpired(long startNanoTime, long expireTime) {
  return nanoTime() - startNanoTime > expireTime;
}
```